### DISCUSSION TITLE
How are you managing quota?

### DISCUSSION THREAD

**Discussion by sdfiokpskiopdf** [mattwright324/youtube-comment-suite#80] [Q&A]
Title says it all. I'm doing a similar project but i'm struggling to manage youtube api quota.

**Comment by mattwright324** [comment]
Unfortunately, I'd assume the problem you are facing is a project quota of 10,000 units and afaik there isn't too much you can do to get a bigger one... aside from trying to request a bigger quota. I am sort of lucky that this project was created when the quota limit for new projects was 1,000,000 units and its kind of grandfathered in. I have some more detail about this in issue #5.

As for how I'm actually managing the quota for this project, I've recently started trying to do more for this issue in particular. How I'm managing quota usage:

- Making it more efficient in the API calls it needs to make
    - Grabbing replies with commentThreads.list with part 'replies' which returns up to 5 replies with each thread. Won't need to make comments.list calls to get replies for threads with 5 or fewer replies.
    - Grabbing the maximum results from each call (50 or 100)
    - Comments and replies have enough channel info to not need to call channels.list. Only need to call channels.list for videos.
- Giving the user choice of what they want to download
    - Maximum API pages for commentThreads / replies
    - Option to stop on a video when it hits a page of comments the app has already downloaded (no new comment ids)
- Other
    - When download finished, telling the user how much estimated quota was used
    - In-app modal with detail about the daily quota reset, when that is in the user's system timezone, and how long until then.

**Reply by sdfiokpskiopdf** [reply]
Thanks for the reply! I'm going to try to implement some of this stuff into my project. Nice work.
